collabcloud/project-collabcloud
# Col-1073 Update GitHub auth

Changelog:
- Moved all OAuth variables (client ID and client secret) into .env files for both client and server
    - For client, make a .env file in client/.env, and add this line to it: 
        `REACT_APP_GITHUB_CLIENT_ID=<client-id-goes-here>`
- Change path of localhost:3000/register2 to now lead to localhost:3000/register/authenticated. This is the URL that GitHub redirects to after the user authorizes the application
- Styling changes in Login and Register pages. Also added tooltips (eg. hovering over "email" field explains to user why we need their email)
- Minor refactoring

Testing:
- Paste the values of `Client ID` and `Client Secret` from [here](https://github.com/organizations/collabcloud/settings/applications/) into you server/config/.env
- Paste JUST `Client ID` into client/.env
- Start the application, and see if you can still register

Extra Note:
Adding an .env file in the client still[ does not prevent users from digging out the client ID if they REALLY](https://create-react-app.dev/docs/adding-custom-environment-variables/) want to. From some quick Googling, it seems like the worst an attacker could do with our client ID is for phishing.

Closes #46 
Closes #73
